packages/sku/config/typescript/tsconfig.js

```diff
@@ -24,14 +24,23 @@ module.exports = () => {
       // Fixes https://github.com/cypress-io/cypress/issues/1087
       skipLibCheck: true,
 
-      esModuleInterop: true,
+      // When dependencies have `type: module` TypeScript doesn't know how to resolve them,
+      // so we need to set this to `bundler` or `node`
+      moduleResolution: 'bundler',
+
+      // resolution-related
       allowSyntheticDefaultImports: true,
+      esModuleInterop: true,
+      isolatedModules: true,
       resolveJsonModule: true,
-      noUnusedLocals: true,
+
+      // misc
       strict: true,
-      jsx: 'preserve',
-      lib: ['dom', 'es2019'],
-      target: 'es5',
+      forceConsistentCasingInFileNames: true,
+      noUnusedLocals: true,
+      jsx: 'react-jsx',
+      lib: ['dom', 'es2022'],
+      target: 'es2022',
```

Comment:
Babel takes care of this via `babel-preset-env` + `browserslist-config-seek`

Reply:
Doesn't `ESNext` have features that are still in proposal stages? I guess it's fine, but it also would require us to keep babel up to date more frequently.

Reply:
I'll leave it, then. This PR has ballooned out of proportion.